lib/ansible/modules/cloud/digital_ocean/digital_ocean_snapshot_facts.py

```diff
@@ -0,0 +1,176 @@
+  digital_ocean_snapshot_facts:
+    snapshot_type: all
+    oauth_token: "{{ oauth_token }}"
+
+- name: Gather facts about droplet snapshots
+  digital_ocean_snapshot_facts:
+    snapshot_type: droplet
+    oauth_token: "{{ oauth_token }}"
+
+- name: Gather facts about volume snapshots
+  digital_ocean_snapshot_facts:
+    snapshot_type: volume
+    oauth_token: "{{ oauth_token }}"
+
+- name: Gather facts about snapshot by snapshot id
+  digital_ocean_snapshot_facts:
+    snapshot_type: by_id
+    snapshot_id: 123123123
+    oauth_token: "{{ oauth_token }}"
+'''
+
+
+RETURN = '''
+data:
+    description: DigitalOcean snapshot facts
+    returned: success
+    type: list
+    sample: [
+        {
+            "id": "4f60fc64-85d1-11e6-a004-000f53315871",
+            "name": "big-data-snapshot1",
+            "regions": [
+                "nyc1"
+            ],
+            "created_at": "2016-09-28T23:14:30Z",
+            "resource_id": "89bcc42f-85cf-11e6-a004-000f53315871",
+            "resource_type": "volume",
+            "min_disk_size": 10,
+            "size_gigabytes": 0
+        },
+    ]
+'''
+
+from traceback import format_exc
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.digital_ocean import DigitalOceanHelper
+from ansible.module_utils._text import to_native
+
+
+def core(module):
+    snapshot_type = module.params['snapshot_type']
+
+    rest = DigitalOceanHelper(module)
+
+    # Check if api_token is valid or not
+    response = rest.get('account')
+    if response.status_code == 401:
+        module.fail_json(msg='Failed to login using api_token, please verify '
+                             'validity of api_token')
+    base_url = 'snapshots'
+    snapshot = []
+    status_code = 400
+
+    if snapshot_type == 'by_id':
+        base_url += "/{0}".format(module.params.get('snapshot_id'))
+        response = rest.get(base_url)
+        status_code = response.status_code
+        snapshot.extend(response.json["snapshot"])
+    else:
+        if snapshot_type == 'droplet':
+            base_url += "?resource_type=droplet&"
+        elif snapshot_type == 'volume':
+            base_url += "?resource_type=volume&"
+
+        page = 1
+        has_next = True
+        while has_next or status_code != 200:
+            required_url = "{0}page={1}&per_page=20".format(base_url, page)
+            response = rest.get(required_url)
+            status_code = response.status_code
+            # stop if any error during pagination
+            if 200 != status_code:
+                break
+            page += 1
+            snapshot.extend(response.json["snapshots"])
+            has_next = "pages" in response.json["links"] and "next" in response.json["links"]["pages"]
+
+    if status_code != 200:
+        module.fail_json(msg="Failed to fetch snapshot facts due to error : %s" % response.json['message'])
+
+    module.exit_json(changed=False, data=snapshot)
+
+
+def main():
+    module = AnsibleModule(
```

Comment:
A few days ago **module_utils/digital_ocean.py** (Ansible Digital Ocean utils) was updated in PR #35934 which introduced generic configurations and documentation that can be reused, namely this function **digital_ocean_argument_spec**(https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/digital_ocean.py)